### DISCUSSION TITLE
Time Zone settings in ReflexSOAR

### DISCUSSION THREAD

**Discussion by ar3diu** [reflexsoar/reflex-api#219] [Q&A] [answered]
Is there a such a setting?

I noticed a +2 difference when created a Case (my timezone is UTC+3):
![image](https://user-images.githubusercontent.com/61625853/176872255-bc5f5a67-78ad-4a23-a370-61ae4cdf916f.png)

Same in the case comments:
![image](https://user-images.githubusercontent.com/61625853/176872685-88382280-5e52-4da3-9a1e-cc3f91596faa.png)

**Comment by n3tsurge** [comment]
The UI library we use is supposed to handle this automatically but I will check it out

Brian Carroll
________________________________
From: Andrei R ***@***.***>
Sent: Friday, July 1, 2022 5:56:31 AM
To: reflexsoar/reflex-api ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [reflexsoar/reflex-api] Time Zone settings in ReflexSOAR (Discussion #219)


Is there a such a setting?

I noticed a +2 difference when created a Case (my timezone is UTC+3):
[image]<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fuser-images.githubusercontent.com%2F61625853%2F176872255-bc5f5a67-78ad-4a23-a370-61ae4cdf916f.png&data=05%7C01%7Cbcarroll%40zeroonesecurity.com%7C6e4c0adc9bb44555260108da5b47fa20%7Ca68b09e0c76f40ccb262ae8575779491%7C0%7C0%7C637922662416757574%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=g5Wl82qJ3IbxjnpHJqNNUCQK%2FN7oiuvuGiZHBP770qo%3D&reserved=0>

**Reply by ar3diu** [reply]
Thanks, Brian!

**Comment by n3tsurge** [comment, marked as answer]
Time issues may explain your MFA issue as well, we store all times in UTC though so it's timezone neutral and shouldn't matter to your authenticator but it's a thought

Brian Carroll
________________________________

**Reply by ar3diu** [reply]
Is there anything I can look at, maybe check some possible misconfigs on my end?

**Reply by ar3diu** [reply]
Also, I can't close any cases due to the same timezone mismatch.

Adding the comment when closing the case...
![image](https://user-images.githubusercontent.com/61625853/176896516-2945703f-b3bd-4366-9be3-8d194b84036e.png)

Checking its status:
![image](https://user-images.githubusercontent.com/61625853/176896637-c6da7064-74de-466d-ace0-6520022b7b36.png)

**Reply by n3tsurge** [reply]
That might be something else, if you hard refresh does it still show In Progress or Closed?

**Reply by n3tsurge** [reply]
Also I can confirm that the UI component should be respecting the Timezone your machine has set so not sure how you're seeing a time drift in the UI like that.  Let me think about how we can troubleshoot that.

![image](https://user-images.githubusercontent.com/1574741/176898267-25b32a26-9084-4bb3-91c9-a74887b20230.png)

**Reply by ar3diu** [reply]
The cause may be a lack of time synchronization on my ReflexSOAR server. So ignore the issue raised here until I solve that. 
`System clock synchronized: no`
`Jul 01 15:59:24 hostname systemd-timesyncd[1835851]: Timed out waiting for reply from X.X.X.X:123 (redacted).`

Thanks for looking into this anyway! I really apreciate!

**Reply by n3tsurge** [reply]
Time being out of sync makes sense in both the TZ issue and MFA issue

**Reply by ar3diu** [reply]
Hi Brian,

I fixed the timezone issue and you were right: the MFA one was related to this as well. Now it works as expected.
